pkg/trace/agent/agent.go

```diff
@@ -623,13 +624,20 @@ func (a *Agent) getAnalyzedEvents(pt *traceutil.ProcessedTrace, ts *info.TagStat
 // runSamplers runs the agent's configured samplers on pt and returns the sampling decision along
 // with the sampling rate.
 //
-// The rare sampler is run first, catching all rare traces early. If the probabilistic sampler is
+// If the agent is set as Error Tracking Standalone, only the ErrorSampler is run (other samplers are bypassed).
+// Otherwise, the rare sampler is run first, catching all rare traces early. If the probabilistic sampler is
 // enabled, it is run on the trace, followed by the error sampler. Otherwise, If the trace has a
 // priority set, the sampling priority is used with the Priority Sampler. When there is no priority
 // set, the NoPrioritySampler is run. Finally, if the trace has not been sampled by the other
 // samplers, the error sampler is run.
 func (a *Agent) runSamplers(now time.Time, ts *info.TagStats, pt traceutil.ProcessedTrace) (keep bool, checkAnalyticsEvents bool) {
-	// run this early to make sure the signature gets counted by the RareSampler.
+	// ETS: chunks that don't contain errors (or spans with exception span events) are all dropped.
+	if a.conf.ErrorTrackingStandalone && traceContainsError(pt.TraceChunk.Spans, true) {
```

Comment:
If `ErrorTrackingStandalone` is enabled, but no error is found on the trace, we will proceed to the other samplers logic. Not sure this is what we want, or what the documentation says? 🤔

Reply:
No that's a mistake 😬 
Fixed in https://github.com/DataDog/datadog-agent/pull/30065/commits/68a8e33893890844f7fef95896c37baea22fe16a